hbase-mapreduce/src/main/java/org/apache/hadoop/hbase/mapreduce/WALInputFormat.java

```diff
@@ -217,8 +217,9 @@ public boolean nextKeyValue() throws IOException, InterruptedException {
         }
         return res;
       } catch (IOException e) {
-        Path archivedLog = AbstractFSWALProvider.getArchivedLogPath(logFile, conf);
-        if (logFile != archivedLog) {
+        Path archivedLog = AbstractFSWALProvider.findArchivedLog(logFile, conf);
+        // archivedLog can be null if unable to locate in archiveDir.
+        if (archivedLog != null && logFile != archivedLog) {
```

Comment:
Should this be `!(logFile.equals(archivedLog)) `or is object equality good enough here? As these are paths there might be more than one object instance (so different regards object equality) but representing equivalent strings?

Reply:
We just use != before the PR, and I think the assumption here is that, the method called here will just return the logFile parameter directly, so != is enough.

Looking at the name 'findArchivedLog', and also the implementation, I do not think we still need to test != here, aas we will never return the logFile itself now. WDYT @shahrs87 ?

Reply:
> Looking at the name 'findArchivedLog', and also the implementation, I do not think we still need to test != here, aas we will never return the logFile itself now. WDYT @shahrs87 ?

Yes, it makes sense that we will never return the logFile again. Removed all the places where  we were comparing logFile with the returned path.